compiler/cpp/src/util/JSONParser.cpp

```diff
@@ -0,0 +1,41 @@
+//
+// Created by Marc on 24.05.2021.
+//
+
+#include "JSONParser.h"
+
+json JSONParser::getJSONValueFromKey(const std::unique_ptr<KeyExprAST>& key) {
+    json dataTmp = data;
+    for (const std::unique_ptr<IdentifierExprAST>& identifier : key->getIdentifiers()) {
+        std::string identifierName = identifier->getName();
+        int identifierIndex = identifier->getIndex();
+        if (!dataTmp.contains(identifierName))
+            throw std::runtime_error("Identifier " + identifierName + " does not exist in data input");
```

Comment:
Replace with custom error message

Reply:
Yes. This will happen in following prs, where I write tests for the individual compiler components.